### PR TITLE
[enhancement](cloud) reduce rpc times in CloudGlobalTransactionMgr when calc delete bitmap for mow

### PR DESCRIPTION
**Pull request by xy720**
## Proposed changes

Issue Number: close #xxx

Use batch mode to get partitions' visible versions to reduce rpc times

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by xy720** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/cloud/transaction/CloudGlobalTransactionMgr.java [hunk 1]
```diff
@@ -548,11 +548,25 @@ private void getPartitionInfo(List<OlapTable> tableList,
 
     private Map<Long, Long> getPartitionVersions(Map<Long, Partition> partitionMap) {
         Map<Long, Long> partitionToVersions = Maps.newHashMap();
+        List<CloudPartition> cloudPartitions = Lists.newArrayList();
         partitionMap.forEach((key, value) -> {
-            long visibleVersion = value.getVisibleVersion();
-            long newVersion = visibleVersion <= 0 ? 2 : visibleVersion + 1;
-            partitionToVersions.put(key, newVersion);
+            if (value instanceof CloudPartition) {
```
Comment (liaoxin01) [inline fe/fe-core/src/main/java/org/apache/doris/cloud/transaction/CloudGlobalTransactionMgr.java:553]: It should be CloudPartition all when come to here.

File: fe/fe-core/src/main/java/org/apache/doris/cloud/transaction/CloudGlobalTransactionMgr.java [hunk 2]
```diff
@@ -548,11 +548,25 @@ private void getPartitionInfo(List<OlapTable> tableList,
...
+                cloudPartitions.add((CloudPartition) value);
+            } else {
+                long visibleVersion = value.getVisibleVersion();
+                long newVersion = visibleVersion <= 0 ? 2 : visibleVersion + 1;
+                partitionToVersions.put(key, newVersion);
+            }
         });
+        try {
+            List<Long> visibleVersion = CloudPartition.getSnapshotVisibleVersion(cloudPartitions);
```
Comment (liaoxin01) [inline fe/fe-core/src/main/java/org/apache/doris/cloud/transaction/CloudGlobalTransactionMgr.java:562]: Can we obtain the latest version using getSnapshotVisibleVersion？

Comment (xy720) [reply fe/fe-core/src/main/java/org/apache/doris/cloud/transaction/CloudGlobalTransactionMgr.java:556]: 是的，你问的很对。这里不能总是获取最新的visible version。getSnapshotVisibleVersion方法调用的meta service rpc方法可见meta_service.cpp的batch_get_version方法，里面get value时使用的是snapshot read。snapshot read与普通read的不同之处在于，允许其他并发的事务修改它们读取的值，而普通read在这种情况下会导致冲突并返回错误。因此这里返回的version可能并不是最新的。

但是这里原来的代码逻辑分成了多次rpc去获取visible version，所以理论上来说也不能获取最新的visible version，因为别的事务commit时可能会修改并让分区的visible version加1。

Comment (xy720) [reply fe/fe-core/src/main/java/org/apache/doris/cloud/transaction/CloudGlobalTransactionMgr.java:556]: https://apple.github.io/foundationdb/developer-guide.html#snapshot-isolation

**Comment by xy720** [comment]
run buildall

**Pull request closed without merging**